core/src/main/java/com/microsoft/applicationinsights/internal/heartbeat/HeartBeatProvider.java

```diff
@@ -0,0 +1,298 @@
+    }
+    return setResult;
+  }
+
+  @Override
+  public boolean isHeartBeatEnabled() {
+    return isEnabled;
+  }
+
+  @Override
+  public void setHeartBeatEnabled(boolean isEnabled) {
+    this.isEnabled = isEnabled;
+  }
+
+  @Override
+  public List<String> getExcludedHeartBeatPropertyProviders() {
+    return this.disabledHeartBeatPropertiesProviders;
+  }
+
+  @Override
+  public void setExcludedHeartBeatPropertyProviders(
+      List<String> excludedHeartBeatPropertyProviders) {
+    this.disabledHeartBeatPropertiesProviders = excludedHeartBeatPropertyProviders;
+  }
+
+  @Override
+  public long getHeartBeatInterval() {
+    return this.interval;
+  }
+
+  @Override
+  public void setHeartBeatInterval(long timeUnit) {
+    // user set time unit in seconds
+    if (timeUnit <= HeartBeatProviderInterface.MINIMUM_HEARTBEAT_INTERVAL) {
+      this.interval = HeartBeatProviderInterface.MINIMUM_HEARTBEAT_INTERVAL;
+    }
+    else {
+      this.interval = timeUnit;
+    }
+  }
+
+  @Override
+  public List<String> getExcludedHeartBeatProperties() {
+    return this.disableDefaultProperties;
+  }
+
+  @Override
+  public void setExcludedHeartBeatProperties(List<String> excludedHeartBeatProperties) {
+    this.disableDefaultProperties = excludedHeartBeatProperties;
+  }
+
+  @Override
+  public void stop(long timeout, TimeUnit timeUnit) {
+    executorService.shutdown();
+    scheduledExecutorService.shutdown();
+    try {
+      executorService.awaitTermination(timeout, timeUnit);
+      scheduledExecutorService.awaitTermination(timeout, timeUnit);
+    }
+    catch (InterruptedException e) {
+      InternalLogger.INSTANCE.warn("unable to successfully terminate heartbeat module, "
+          + "encountered and exception with stacktrace, %s", ExceptionUtils.getStackTrace(e));
+    }
+  }
+
+  /**
+   * Send the heartbeat item synchronously to application insights backend.
+   */
+  private void send() {
+
+    synchronized (lock) {
+      MetricTelemetry telemetry = (MetricTelemetry)gatherData();
+      telemetry.getContext().getOperation().setSyntheticSource(HEARTBEAT_SYNTHETIC_METRIC_NAME);
+      telemetryClient.trackMetric(telemetry);
+      heartbeatsSent.incrementAndGet();
+      InternalLogger.INSTANCE.trace("No of heartbeats sent, %s", heartbeatsSent.get());
+    }
+
+  }
+
+  /**
+   * Creates and returns the heartbeat telemetry.
+   * @return Metric Telemetry which represent heartbeat.
+   */
+  private Telemetry gatherData() {
+
+    MetricTelemetry heartbeat = new MetricTelemetry(HEARTBEAT_SYNTHETIC_METRIC_NAME, 0.0);
+    Map<String, String> property = heartbeat.getProperties();
+    for (Map.Entry<String, HeartBeatPropertyPayload> entry : heartbeatProperties.entrySet()) {
+      property.put(entry.getKey(), entry.getValue().getPayloadValue());
+      double currentValue = heartbeat.getValue();
+      currentValue += entry.getValue().isHealthy() ? 0 : 1;
+      heartbeat.setValue(currentValue);
```

Comment:
https://github.com/Microsoft/ApplicationInsights-dotnet/issues/705 This issue is addressed right?  :)

Reply:
@cijothomas I am not sure. I will have to check this. Most likely not, but I think we can take this in the later iteration given the release time constrain by build. Let me know your thoughts.